packages/amplify_core/lib/src/types/storage/storage_path_from_identity_id.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:meta/meta.dart';
+
+/// {@template amplify_core.storage.storage_path_from_identity_id}
+/// Creates a [StoragePath] from the current user's identityId.
+/// {@endtemplate}
+class StoragePathFromIdentityId implements StoragePath {
```

Comment:
This is private. The type isn't exported from the public API so it would have to be imported from the full file path to be used and consumers would get a lint warning they would need to ignore (`ignore: implementation_imports`).

The only benefit that I can see of marking it as internal is in case we accidentally add it to the public exports. The downside of marking it as internal is that we have to ignore the lint warnings where it is used in the storage package. If we are concerned about accidentally exporting something that should not be public I think a better alternative is to look into tools that can check for API changes on each PR. 

The pattern that Amplify Flutter seems to follow currently is to only mark something as internal if it is being exported as part of the public API. If you search for `ignore: implementation_imports` you can find ~100 references to classes/constants that are not exported from the public API and are not marked as internal.

re: combining into a single class - We would need a way to tell what type of StoragePath this is so that we can only fetch identity Id for paths that require it. Currently we use to the class type to tell what type of StoragePath it is. So we would need to add a public (marked as @internal) attribute to indicate what type of storage path this is. I think a separate class is more scalable. It is possible/likely that we will have other types of StoragePaths in the future. Combining them all into a single class will be hard to maintain.

Reply:
Thank you, Jordan, for the clarification. I overlooked the fact that it wasn't exported from storage_type.dart.